data/mods/gen2stadium2/scripts.ts

```diff
@@ -329,27 +328,51 @@ export const Scripts: ModdedBattleScriptsData = {
 				level = move.allies[0].level;
 			}
 
-			let attacker = pokemon;
-			const defender = target;
-			if (move.useTargetOffensive) attacker = target;
-			let atkType: StatIDExceptHP = (move.category === 'Physical') ? 'atk' : 'spa';
-			const defType: StatIDExceptHP = (move.defensiveCategory === 'Physical') ? 'def' : 'spd';
-			if (move.useSourceDefensiveAsOffensive) atkType = defType;
+			const attacker =
+				(move.useBaseOffensiveStatAndBoosts && move.useBaseOffensiveStatAndBoosts.includes('target')) ? target : pokemon;
+			const defender =
+				(move.useBaseDefensiveStatAndBoosts && move.useBaseDefensiveStatAndBoosts.includes('source')) ? pokemon : target;
+
+			let atkType: AllStatIDs = move.category === 'Physical' ? 'atk' : 'spa';
+			if (move.useBaseOffensiveStatAndBoosts) {
+				atkType = move.useBaseOffensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
+			let defType: AllStatIDs = move.category === 'Physical' ? 'def' : 'spd';
+			if (move.useBaseDefensiveStatAndBoosts) {
+				defType = move.useBaseDefensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
 			let unboosted = false;
 			let noburndrop = false;
 
 			if (isCrit) {
 				if (!suppressMessages) this.battle.add('-crit', target);
 				// Stat level modifications are ignored if they are neutral to or favour the defender.
 				// Reflect and Light Screen defensive boosts are only ignored if stat level modifications were also ignored as a result of that.
-				if (attacker.boosts[atkType] <= defender.boosts[defType]) {
+				if ((atkType === 'hp' || atkType === 'currenthp') ? 0 : attacker.boosts[atkType] <=
+					((defType === 'hp' || defType === 'currenthp') ? 0 : defender.boosts[defType])) {
 					unboosted = true;
 					noburndrop = true;
 				}
 			}
-			// Get stats now.
-			let attack = attacker.getStat(atkType, unboosted, noburndrop);
-			let defense = defender.getStat(defType, unboosted);
+
+			let attack;
+			let defense;
+
+			if (atkType === 'hp') {
+				attack = attacker.maxhp;
+			} else if (atkType === 'currenthp') {
+				attack = attacker.hp;
+			} else {
+				attack = attacker.getStat(atkType, unboosted, noburndrop);
+			}
+
+			if (defType === 'hp') {
+				defense = defender.maxhp;
+			} else if (defType === 'currenthp') {
+				defense = defender.hp;
+			} else {
+				defense = defender.getStat(defType, unboosted);
+			}
```

Comment:
ditto

Reply:
(same comment as above)
I assume you are referencing optional chaining? It will not work here due to what is needed to be done for the alternative cases. If I am just missing something please let me know. I feel like this is the better and easier to read solution to the problem.

Reply:
inline if

Reply:
Fixed, thank you.